Quirk: Light Sleeper

V2LenKagamine opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
_**HEY! GIVE ME FEEDBACK! I'm here to put out an idea, you're here to help me refine it into something worthy of being in the game!**_

This simple pull request does 2 things:
Adds the quirk ~~"Insomniac"~~ "Light Sleeper" reducing sleep and unconscious times by 25%
Changes "Heavy Sleeper" to no longer be random, and always increase sleep and unconscious times by 25%
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Ever wonder why Heavy Sleeper didn't have a POSITIVE variant? I did.

I think we could use more quirks, while I would prefer them to be more game changing, the community appears to only want RP related quirks. So I made a less impactful quirk. Now you can be even more of a snowflake!

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Added Quirk: Light Sleeper
tweak: "Heavy Sleeper" now always increases Sleep and Unconscious times by 25% instead of being between 25% and 30%.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

V2LenKagamine committed:
yote

V2LenKagamine committed:
Adds quirk "Insomniac"

Adds the Quirk: Insomniac

V2LenKagamine committed:
Revert "yote"

This reverts commit f766414b2b059b6994f2e8747e5e10a52ac2b3c2.

ATH1909 commented:
Make them mutually exclusive; taking both heavy sleeper AND insomniac gives you a 6.25% reduction in knockout times for a net quirk cost of 0 points.

RaveRadbury commented:
> Make them mutually exclusive; taking both heavy sleeper AND insomniac gives you a 6.25% reduction in knockout times for a net quirk cost of 0 points.

Alternatively rebalance them so you get a 6.25% _increase_ and greedy fools are punished.

anconfuzedrock commented:
alternatively don't add this, this is too close to the "no combat quirks" thing which I May not agree with but every single fucking powergamer will get reduced sleep times and, hell, everyone, since being able to wake up faster is so convenient.

V2LenKagamine committed:
Powergame-b-gone

Perk now punishes powergamers by only giving the +25% from Heavy Sleeper without the -25% from Insomniac.

tralezab commented:
So weird seeing "Insomniac" as a good perk and "Heavy Sleeper" as a bad one

ATH1909 commented:
> alternatively don't add this, this is too close to the "no combat quirks" thing which I May not agree with but every single fucking powergamer will get reduced sleep times and, hell, everyone, since being able to wake up faster is so convenient.

Then what the hell _are_ we supposed to spend our quirk points on? Everyone seems to protest the addition of anything that might be worth taking a negative quirk (other than light drinker; that's practically a freebie) for.

Ratvar commented:
As Oranges said, "Quirks are meant to be RP flavouring with minor in game impacts", not for buffing your character up. Yes, not for being *worth* to take, but for being snowflake. Like, can you say why you wouldn't take this trait?

Mickyan commented on code/modules/mob/living/carbon/human/status_procs.dm:
```diff
@@ -18,12 +18,16 @@
 	if(has_trait(TRAIT_HEAVY_SLEEPER))
-		amount *= rand(1.25, 1.3)
+		amount *= 1.25
+	if(has_trait(TRAIT_LIGHT_SLEEPER && !TRAIT_HEAVY_SLEEPER))
```
we have quirk_blacklist for this

dcaine00 replied:
Sorry, Im a bit new to coding and the codebase, and this was the easiest and quickest solution I could think of.

V2LenKagamine replied:
Fixed!

anconfuzedrock commented:
just because you don't like the current quirks doesnt mean they're useless. You're probably a validhunter who plays a byond game for the combat, so I recommend empath: that way you can perform disgusting antag checks by forcefeeding people urinal cakes and seeing if that makes their sanity go down too much. Otherwise, climbing tables faster or being able to hide in a locker while handcuffed when you're being dragged off to brig for smashing random windows with a fire axe and the officer slips and you run off faster than your mother to an abortion clinic in an alternate reality where she can see what a disappointment you're going to be, could have use. Or- here's an idea- don't take quirks at all if you don't like the upsides, nobody is forcing you to take the downsides.

RaveRadbury commented:
The funny thing is that Heavy Sleeper is already a power-game quirk because a lot of people believe that once you've gone horizontal the fight is over anyway. So in their minds it's already free points.

OverlyGeneric commented:
> The funny thing is that Heavy Sleeper is already a power-game quirk because a lot of people believe that once you've gone horizontal the fight is over anyway. So in their minds it's already free points.

It also increases the time for regenerative coma to heal you. I think.

Militaires commented:
look in quirks.dm subsystem folder, you'll find the blacklist there

V2LenKagamine committed:
Namechange

Changed name from "Insomniac" to "Light Sleeper" to avoid confusion
Added to quirk Blacklist

PKPenguin321 commented:
@optimumtact combat oriented quirk detected

optimumtact commented:
what things in combat make you go to sleep?

PKPenguin321 commented:
off the top of my head:
syringe guns with sleep chems (very common)
sleep chems in general
abductor sleep prod
boxing gloves KO punch

Militaires commented:
honestly if someone is gonna kill you when you're asleep a 25% shorter time wont make a difference, so you can actually just not consider it as a combat quirk

Shadowflame909 commented:
How do you get up 25% faster anyways when Morphine is inside of you. Wouldn't the chemical be forcing you asleep anyways?

anconfuzedrock commented:
@optimumtact most importantly explosions though, especially fireballs. this would make it more feasible to get up from a fireball or welderbomb and put yourself out.

Either way, this quirk is absurdly good and it will be unless it's like 3 points at minimum, I'm shocked and apalled this guy looked at the feedback and thought "yep what this needs is a name change." Like I couldn't imagine shitting out this many prs with so much effort put into them yet so little thought.

Tlaltecuhtli commented:
sleeping from chems isnt actually important as you ll be asleep as long as you have the chem in body (it only affects the last few seconds after it depletes )

real things affected by heavy sleeper/insomniac

cryo torture pod

rng explosion knockdown which i think should be removed

anconfuzedrock commented:
I swear to god if I'm a cook and this fucking assistant with fast tableclimbing wakes up even faster after they get knocked out with CC and pours acid on my spaghetti I am going to lose it

ExcessiveUseOfCobblestone commented:
The fact unconscious and sleep are 2 different procs leads me to believe at least 1 has combat usage if not both.

ExcessiveUseOfCobblestone commented:
https://github.com/tgstation/tgstation/blob/c73cf825be70acde4580410cd4f11cd19fe72248/code/game/gamemodes/cult/cult.dm#L110-L111

https://github.com/tgstation/tgstation/blob/c73cf825be70acde4580410cd4f11cd19fe72248/code/game/gamemodes/cult/cult.dm#L121-L122

https://github.com/tgstation/tgstation/blob/f642c66986fee91548f2110185bc2057ee0e2b9d/code/modules/spells/spell_types/hivemind.dm#L201

https://github.com/tgstation/tgstation/blob/55a768fef109af873291a4383e1b735a5e5969c5/code/modules/hydroponics/grown/nettle.dm#L113

just a few

anconfuzedrock commented:
!!Mindswap!!

necromanceranne commented:
CQC knockout. But frankly if someone has you in CQC stunlock hell then likely you're already dead.

ExcessiveUseOfCobblestone commented:
The implication is that if they can knock you down they can follow up, which isn't always true (at least to the point where this is negligible for considering this combat-oriented).

anconfuzedrock commented:
I greatly struggle to think of a situation where you're knocked out that isn't combat oriented besides *sleep which is lol you're trying to sleep

CRITAWAKETS commented:
Honestly, i would not have this quirk because its useless. Most situations where you get knocked out are you being unconcious, not asleep. This doesnt help agaisnt sleep chems, infact it can make it worse with some doses because you can go back to sleep a second time sooner.

ExcessiveUseOfCobblestone commented:
Basically a stun-modifying quirk which is not allowed.
